### DISCUSSION TITLE
Scaling from 0 pods.

### DISCUSSION THREAD

**Discussion by shashanksdixit** [kedacore/keda#1626] [Q&A / Need Help] [answered]
In our case, the minimum number of pods are given as 0. So initially, when the messageCount= 5 (default), then the pods are created. The number of pods will increase as the number of messages will increase. 

I have a question when the number of pods are 0. i.e. there are no pods. Keda will create the pod when there are messages.
If for some reason, the messageCount is not achieved e.g. messageCount will be 2, then will Keda create pod. Or Keda will wait till the count become 5? 

If we want to scale the pods for each 20 active messages in the queue (messageCount = 20), but we also want if there are no pods (initially), then on first message, the pod should be created. Is this achievable?

**Comment by tomkerkhove** [comment, marked as answer]
Normally it should spin up a pod as soon as there are messages. THe message count is more the threshold for adding new instances for the 1 -> n scaling.

So what you are looking for, is what we do. But you can easily verify it!

**Reply by shashanksdixit** [reply]
Thanks @tomkerkhove for the reply.
I had verified it earlier. But, I wanted to confirm my understanding.

**Reply by tomkerkhove** [reply]
Happy to help!
